stdlib/Test/src/logging.jl

```diff
@@ -29,6 +42,46 @@ mutable struct TestLogger <: AbstractLogger
     message_limits::Dict{Any,Int}
 end
 
+"""
+    TestLogger(; min_level=Info, catch_exceptions=false)
+
+Create a `TestLogger` which captures logged messages in its `logs::Vector{LogRecord}` field.
+
+See also: [`LogRecord`](@ref).
+
+## Example
+
+```jldoctest
+julia> using Test, Logging
+
+julia> f() = @info "Hi" number=5
+f (generic function with 1 method)
+
+julia> test_logger = TestLogger()
+TestLogger(LogRecord[], Info, false, nothing)
+
+julia> with_logger(test_logger) do
+               f()
+               @info "Bye!"
+           end
+
+julia> @test test_logger.logs[1].message == "Hi"
+Test Passed
+  Expression: (test_logger.logs[1]).message == "Hi"
+   Evaluated: "Hi" == "Hi"
```

Comment:
I think this verbose test output have been reverted on master.

Reply:
ah, that definitely would've confused me, thanks!